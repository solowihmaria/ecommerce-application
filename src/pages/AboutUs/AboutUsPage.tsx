import { TeamIntro } from '../../components/blocks/TeamIntro';
import styles from './AboutUsPage.module.scss';

export const AboutUsPage = () => (
    <div data-testid="about-us-page" className={styles.aboutUsPageContainer}>
        <TeamIntro />
    </div>
);
