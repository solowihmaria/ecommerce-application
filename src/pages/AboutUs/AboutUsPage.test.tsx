import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { AboutUsPage } from './AboutUsPage';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../components/blocks/TeamIntro/TeamIntro', () => ({
    TeamIntro: () => <div data-testid="test-id-team-intro">TeamIntro Mock</div>,
}));

describe('<AboutUsPage/> component', () => {
    test('Render AboutUsPage component', () => {
        render(
            <MemoryRouter>
                <AboutUsPage />
            </MemoryRouter>
        );

        const aboutPage = screen.getByTestId('about-us-page');
        expect(aboutPage).toBeInTheDocument();

        expect(screen.getByTestId('test-id-team-intro')).toBeInTheDocument();
        expect(aboutPage).toMatchSnapshot();
    });
});
