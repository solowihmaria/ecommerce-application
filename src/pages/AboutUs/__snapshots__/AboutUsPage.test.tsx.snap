// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutUsPage/> component Render AboutUsPage component 1`] = `
<div
  class="aboutUsPageContainer"
  data-testid="about-us-page"
>
  <div
    data-testid="test-id-team-intro"
  >
    TeamIntro Mock
  </div>
</div>
`;
